core/components/atoms/chip/__tests__/Chip.test.tsx

```diff
@@ -138,4 +138,34 @@ describe('Chip component', () => {
     );
     expect(queryByTestId('DesignSystem-Chip--GenericChip')).toHaveClass('Chip-selection--selectedDisabled');
   });
+
+  it('renders chip component with custom label', () => {
+    const labelText = 'Chip';
+    const customLabel = <Text>{labelText}</Text>;
+    const { queryByTestId } = render(<Chip label={customLabel} name={labelText} />);
+    expect(queryByTestId('DesignSystem-Chip--GenericChip')).toBeInTheDocument();
+    expect(queryByTestId('DesignSystem-Text')).toHaveTextContent(labelText);
+  });
+
+  it('renders chip component with multiple labels', () => {
+    const labelText1 = 'Chip1';
+    const labelText2 = 'Chip2';
+    const customLabel = (
+      <span>
+        <Text>{labelText1}</Text>
+        <Text>{labelText2}</Text>
+      </span>
+    );
+    const { queryAllByTestId } = render(<Chip label={customLabel} name="Chip" />);
+    const labels = queryAllByTestId('DesignSystem-Text');
+    expect(labels[0]).toHaveTextContent(labelText1);
+    expect(labels[1]).toHaveTextContent(labelText2);
+  });
+
+  it('renders chip component with styled labels', () => {
+    const labelText = 'Chip';
+    const customLabel = <Text color="primary">{labelText}</Text>;
+    const { queryByTestId } = render(<Chip label={customLabel} name={labelText} />);
+    expect(queryByTestId('DesignSystem-Text')).toHaveClass('color-primary');
+  });
```

Comment:
Can we please add a valid test case to check the arguments passed in the `onClose` & `onClick` event handlers?

Reply:
Test cases created